tests/data_context/store/test_checkpoint_store.py

```diff
@@ -350,3 +367,164 @@ def test_serialization_self_check(capsys) -> None:
+
+    with pytest.raises(ge_exceptions.CheckpointNotFoundError) as e:
+        store.delete_checkpoint(name="my_fake_checkpoint")
+
+    assert 'Non-existent Checkpoint configuration named "my_fake_checkpoint".' in str(
+        e.value
+    )
+
+
+@pytest.mark.unit
+def test_get_checkpoint(
+    checkpoint_store_with_mock_backend: Tuple[CheckpointStore, mock.MagicMock],
+    checkpoint_config: dict,
+) -> None:
+    store, mock_backend = checkpoint_store_with_mock_backend
+    mock_backend.get.return_value = checkpoint_config
+
+    checkpoint = store.get_checkpoint(name=checkpoint_config["name"], ge_cloud_id=None)
+
+    actual_checkpoint_config = checkpoint.to_json_dict()
+    for key, val in checkpoint_config.items():
+        assert val == actual_checkpoint_config.get(key)
+
+
+@pytest.mark.unit
+def test_get_checkpoint_with_nonexistent_checkpoint_raises_error(
+    checkpoint_store_with_mock_backend: Tuple[CheckpointStore, mock.MagicMock]
+) -> None:
+    def _raise_key_error(_: Any) -> None:
+        raise ge_exceptions.InvalidKeyError(message="invalid key")
+
+    store, mock_backend = checkpoint_store_with_mock_backend
+    mock_backend.get.side_effect = _raise_key_error
+
+    with pytest.raises(ge_exceptions.CheckpointNotFoundError) as e:
+        store.get_checkpoint(name="my_fake_checkpoint", ge_cloud_id=None)
+
+    assert 'Non-existent Checkpoint configuration named "my_fake_checkpoint".' in str(
+        e.value
+    )
+
+
+@pytest.mark.unit
+def test_get_checkpoint_with_invalid_checkpoint_config_raises_error(
+    checkpoint_store_with_mock_backend: Tuple[CheckpointStore, mock.MagicMock]
+) -> None:
+    def _raise_validation_error(_: Any) -> None:
+        raise ValidationError(message="invalid config")
+
+    store, mock_backend = checkpoint_store_with_mock_backend
+    mock_backend.get.return_value = {"class_name": "Checkpoint"}
+
+    with mock.patch(
+        "great_expectations.data_context.store.CheckpointStore.deserialize",
+        side_effect=_raise_validation_error,
+    ), pytest.raises(ge_exceptions.InvalidCheckpointConfigError) as e:
+        store.get_checkpoint(name="my_fake_checkpoint", ge_cloud_id=None)
+
+    assert "Invalid Checkpoint configuration" in str(e.value)
+
+
+@pytest.mark.unit
+def test_get_checkpoint_with_invalid_legacy_checkpoint_raises_error(
+    checkpoint_store_with_mock_backend: Tuple[CheckpointStore, mock.MagicMock]
+) -> None:
+    store, mock_backend = checkpoint_store_with_mock_backend
+    mock_backend.get.return_value = (
+        CheckpointConfig().to_json_dict()
+    )  # Defaults to empty LegacyCheckpoint
+
+    with pytest.raises(ge_exceptions.CheckpointError) as e:
+        store.get_checkpoint(name="my_checkpoint", ge_cloud_id=None)
+
+    assert (
+        "Attempt to instantiate LegacyCheckpoint with insufficient and/or incorrect arguments"
+        in str(e.value)
+    )
+
+
+@pytest.mark.unit
+def test_add_checkpoint(
```

Comment:
Does it make sense to call this a legacy checkpoint test and create a separate test with non legacy checkpoint?

Reply:
so this particular test doesn't actually have any logical around legacy vs current checkpoints (while the one above did). For that reason, I don't think we need to get as granular and separate it out with more tests.

Let me know if you disagree and I'd be happy to add another test.

Reply:
I think that makes sense. You would only be testing that add_checkpoint was called with a different checkpoint config which shouldn't be different than the test as written.